cmd/influxd/upgrade/upgrade.go

```diff
@@ -44,72 +79,293 @@ var options = struct {
+			Default: defaultSsPath,
+			Desc:    "optional: generated security upgrade script path",
+		},
+		{
+			DestP:   &options.logPath,
+			Flag:    "log-path",
+			Default: filepath.Join(homeOrAnyDir(), "upgrade.log"),
+			Desc:    "optional: custom log file path",
+		},
+	}
 
+	cli.BindOptions(cmd, opts)
 	// add sub commands
-	Command.AddCommand(v1DumpMetaCommand)
-	Command.AddCommand(v2DumpMetaCommand)
+	cmd.AddCommand(v1DumpMetaCommand)
+	cmd.AddCommand(v2DumpMetaCommand)
+	return cmd
 }
 
 type influxDBv1 struct {
 	meta *meta.Client
 }
 
 type influxDBv2 struct {
+	log         *zap.Logger
 	boltClient  *bolt.Client
 	store       *bolt.KVStore
 	kvStore     kv.SchemaStore
 	tenantStore *tenant.Store
 	ts          *tenant.Service
 	dbrpSvc     influxdb.DBRPMappingServiceV2
+	bucketSvc   influxdb.BucketService
 	onboardSvc  influxdb.OnboardingService
 	kvService   *kv.Service
 	meta        *meta.Client
 }
 
-func runUpgradeE(cmd *cobra.Command, args []string) error {
+func (i *influxDBv2) close() error {
+	err := i.meta.Close()
+	if err != nil {
+		return err
+	}
+	err = i.boltClient.Close()
+	if err != nil {
+		return err
+	}
+	err = i.store.Close()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func runUpgradeE(*cobra.Command, []string) error {
 	ctx := context.Background()
+	config := zap.NewProductionConfig()
+	config.OutputPaths = append(config.OutputPaths, options.logPath)
+	config.ErrorOutputPaths = append(config.ErrorOutputPaths, options.logPath)
+	log, err := config.Build()
+	if err != nil {
+		return err
+	}
+	log.Info("starting InfluxDB 1.x upgrade")
+
+	checkParam := func(name, value string) error {
+		if value == "" {
+			return fmt.Errorf("empty or missing mandatory option %s", name)
+		}
+		return nil
+	}
+
+	if err := checkParam("username", options.target.userName); err != nil {
+		return err
+	}
+	if err := checkParam("password", options.target.password); err != nil {
+		return err
+	}
+	if err := checkParam("org", options.target.orgName); err != nil {
+		return err
+	}
+	if err := checkParam("bucket", options.target.bucket); err != nil {
+		return err
+	}
+
+	if options.source.configFile != "" {
+		log.Info("upgrading config file", zap.String("file", options.source.configFile))
+		if _, err := os.Stat(options.source.configFile); err != nil {
+			return err
+		}
+		v1Config, err := upgradeConfig(options.source.configFile, options.target, log)
+		if err != nil {
+			return err
+		}
+		options.source.metaDir = v1Config.Meta.Dir
+		options.source.dataDir = v1Config.Data.Dir
+		options.source.walDir = v1Config.Data.WALDir
+	} else {
+		log.Info("no InfluxDB 1.x config file specified, skipping its upgrade")
+	}
+
+	if err := options.source.checkDirs(); err != nil {
+		return err
+	}
+
+	log.Info("source paths", zap.String("meta", options.source.metaDir), zap.String("data", options.source.dataDir))
+	log.Info("target paths", zap.String("bolt", options.target.boltPath), zap.String("engine", options.target.enginePath))
```

Comment:
nit: Log messages should be complete sentences and start with a capital. Please see the [logging style guide](https://github.com/influxdata/influxdb/blob/master/logger/style_guide.md)